packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -64,7 +234,7 @@ class FlutterErrorDetails {
   /// following the word "thrown", as in "thrown while obtaining the image from
   /// the network" (for the context "while obtaining the image from the
   /// network").
-  final String context;
+  final DiagnosticsNode context;
```

Comment:
this one surprises me. The `context` was intended to just be a string that gets interpolated into another; does it still work well if it's a DiagnosticsNode?

Reply:
Yes. DiagnosticsNode in general and ErrorDescription in particular works great with String interpolation. As long as you wrap each call to string interpolation with an ErrorDescription you will maintain all the nested object references so that they are still clickable in the inspector regardless of how much nested interpolation you did.
Basically you can think of an ErrorDescription as just a String with object references captured for purposes of String interpolation.

Examples of use cases this supports:
```dart
          context: ErrorDescription('during a service extension callback for "$method"'),
          context: ErrorDescription('while notifying listeners for $runtimeType'),
```
Advanced case in debugFillProperties for FlutterErrorDetails where we use this
```
    final DiagnosticsNode verb = ErrorDescription('thrown${ context != null ? ErrorDescription(" $context") : ""}');

```

Reply:
@Hixie does this address your concerns about context being a DiagnosticsNode? Put more concisely: this allows the same interpolation use case and if you use ErrorDescription in every level of interpolation you don't lose the object references.

Reply:
Yeah, that sounds good.